kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/OperatorMain.java

```diff
@@ -66,28 +74,31 @@ void start() {
         operator.installShutdownHook(Duration.ofSeconds(10));
         var registeredController = operator.register(new ProxyReconciler(runtimeDecl()));
         // TODO couple the health of the registeredController to the operator's HTTP healthchecks
+        managementServer.createContext("/metrics", metricsHandler);
```

Comment:
I am remember #1039.   It applies to the Proxy itself, but the same will apply here now it exists. What would you using a custom com.sun.net.httpserver.Filter to reject anything but GET requests?

Reply:
I agree it applies here and was about to say I'd defer it to a separate PR and do both at once. However as this is a simple java server and the proxy one is netty, I think its better to do this one in scope of this PR.

Reply:
I was suggesting using Filter rather than a Handler.    I think if you took that route you could avoid having to add the handler for `/`.

Reply:
I experimented with this. We do avoid the delegation however we still need the createContext and to attach a handler (Partially as we allow the injection of the management server for testing).

If we dynamically register the filter without supplying a `HttpHandler` we get server 500 errors when requesting anything tother the `/metrics`. To avoid that I've added simple handler which returns a 404 for `/` and thus any path that isn't under `/metrics`. Maybe we should do something similar for `/metrics/*` rather than just delegating to the prometheus metrics handler, thats of limited utility.